File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/interactors/CardsInteractor.java
package com.dbottillo.mtgsearchfree.interactors;

import com.dbottillo.mtgsearchfree.model.Deck;
import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.dbottillo.mtgsearchfree.model.MTGSet;
import com.dbottillo.mtgsearchfree.model.SearchParams;

import java.util.List;

import io.reactivex.Observable;

public interface CardsInteractor {

    Observable<List<MTGCard>> loadSet(MTGSet set);

    Observable<int[]> saveAsFavourite(MTGCard card);

    Observable<int[]> removeFromFavourite(MTGCard card);

    Observable<int[]> loadIdFav();

    Observable<List<MTGCard>> getLuckyCards(int howMany);

    Observable<List<MTGCard>> getFavourites();

    Observable<List<MTGCard>> loadDeck(Deck deck);

    Observable<List<MTGCard>> doSearch(SearchParams searchParams);

    Observable<MTGCard> loadCard(int multiverseid);

    Observable<MTGCard> loadOtherSideCard(MTGCard card);
}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/interactors/CardsInteractor.kt
package com.dbottillo.mtgsearchfree.interactors

import com.dbottillo.mtgsearchfree.model.CardsCollection
import com.dbottillo.mtgsearchfree.model.MTGCard
import com.dbottillo.mtgsearchfree.model.MTGSet
import com.dbottillo.mtgsearchfree.model.SearchParams
import io.reactivex.Observable
import io.reactivex.Single

interface CardsInteractor {
    fun loadSet(set: MTGSet): Observable<CardsCollection>
    fun saveAsFavourite(card: MTGCard)
    fun removeFromFavourite(card: MTGCard)
    fun loadIdFav(): Observable<IntArray>
    fun getLuckyCards(howMany: Int): Observable<CardsCollection>
    fun getFavourites(): Observable<List<MTGCard>>
    fun doSearch(searchParams: SearchParams): Observable<CardsCollection>
    fun loadCard(multiverseId: Int): Observable<MTGCard>
    fun loadCardById(id: Int): Single<MTGCard>
    fun loadOtherSideCard(card: MTGCard): Observable<MTGCard>
}